File: examples/TEMPLATE.py
# /// script
# dependencies = ["brahe", "pytest"]
# ///
"""
TODO: Write a brief description of what this example demonstrates.
"""

import pytest

# Setup: Define any input parameters
value = 1.0

# Action: Demonstrate the functionality
result = value * 2.0  # Replace with actual brahe function call

# Validation: Assert the result is correct
expected = 2.0
assert result == pytest.approx(expected, abs=1e-10)

print("✓ Example validated successfully!")

File: examples/TEMPLATE.rs
//! Brief description of what this example demonstrates.
//!
//! This template shows the minimal structure for a Rust documentation example.
//! Replace this with actual functionality demonstration.

use approx::assert_abs_diff_eq;
#[allow(unused_imports)]
use brahe::time::{Epoch, TimeSystem};

fn main() {
    // Setup: Define any input parameters
    let value = 1.0;

    // Action: Demonstrate the functionality
    let result = value * 2.0; // Replace with actual brahe function call

    // Validation: Assert the result is correct
    let expected = 2.0;
    assert_abs_diff_eq!(result, expected, epsilon = 1e-10);

    println!("✓ Example validated successfully!");
}
